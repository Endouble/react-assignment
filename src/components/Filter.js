import React from 'react';
import PropTypes from 'prop-types';
import { Form, Input, Button, Segment, Header } from 'semantic-ui-react';
import '../styles/Filter.css';

/**
 * Key Press handler
 *
 * Prevents the user to insert not Digit characters in number inputs
 * This is because the API only accepts Integer numbers
 */
const handleKeyPress = (event) => {
    const re = /[0-9]/g;
    if (!re.test(event.key) && event.key !== 'Enter') {
        event.preventDefault();
    }
};

/**
 * Filter Component
 *
 * Render a form that executes the onFilter action (received by props) in the onSubmit event
 * This action may cause the App Component to retrieve filtered data from the API
 */
class Filter extends React.Component {
    constructor(props) {
        super(props);
        this.handleSubmit = this.handleSubmit.bind(this);
        this.beerNameField = React.createRef();
        this.ibuGtField = React.createRef();
        this.abvGtField = React.createRef();
    }

    handleSubmit(event) {
        event.preventDefault();
        const { onFilter } = this.props;
        const filters = {
            beer_name: this.beerNameField.current.inputRef.value,
            ibu_gt: this.ibuGtField.current.inputRef.value,
            abv_gt: this.abvGtField.current.inputRef.value,
        };
        onFilter(filters);
    }

    render() {
        return (
            <Segment className="filters-wrapper">
                <Header as='h3'>Filters</Header>
                <Form onSubmit={this.handleSubmit} role="search">
                    <Form.Group widths='equal'>
                        <Form.Field>
                            <label htmlFor='beerName'>Name</label>
                            <Input placeholder='Name' id='beerName' ref={this.beerNameField} />
                        </Form.Field>
                        <Form.Field>
                            <label htmlFor='beerIbuGt'>IBU Greater Than</label>
                            <Input type='number' placeholder='IBU Greater Than' id='beerIbuGt' ref={this.ibuGtField} onKeyPress={handleKeyPress} />
                        </Form.Field>
                        <Form.Field>
                            <label htmlFor='beerAbvGt'>ABV Greater Than</label>
                            <Input type='number' placeholder='ABV Greater Than' id='beerAbvGt' ref={this.abvGtField} onKeyPress={handleKeyPress} />
                        </Form.Field>
                        <Form.Field className='form-action-container'>
                            <Button primary>Filter</Button>
                        </Form.Field>
                    </Form.Group>
                </Form>
            </Segment>
        );
    }
}

Filter.propTypes = {
    onFilter: PropTypes.func.isRequired,
};

export default Filter;
