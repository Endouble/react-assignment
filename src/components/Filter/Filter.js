import React, { Component } from 'react'

class filterForm extends Component {
  constructor(props) {
    super(props)
    this.state = {
        filterValue: ""
    }
  }
  
  handleChange = (e) => {
    this.setState({
      filterValue: e.target.value
    })
    this.props.onChange(e.target.value)
  }
  
  render() {
    return (
      <div>
        <label htmlFor="filter">Filter: </label>
        <input type="text" id="filter" 
          value={this.state.filterValue} 
          onChange={this.handleChange}/>
      </div>
      )
  }
}

export default filterForm