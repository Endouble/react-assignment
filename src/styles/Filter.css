.filters-wrapper.ui.segment {
    margin-top: 14px;
}

.filters-wrapper .form-action-container {
    display: flex;
    justify-content: flex-end;
    flex-direction: column;
}