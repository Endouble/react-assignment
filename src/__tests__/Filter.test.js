import React from 'react';
import { shallow } from 'enzyme';
import Filter from '../components/Filter';

describe('Filter Unit Test', () => {
    let props;
    let shallowedFilter;

    const getFilter = () => {
        if (!shallowedFilter) {
            shallowedFilter = shallow(<Filter {...props} />);
        }
        return shallowedFilter;
    };

    beforeEach(() => {
        props = {
            onFilter: jest.fn()
        };
    });

    test('Component should render without crashing', () => {
        expect(getFilter()).toBeDefined();
    });

    test('Component should render correctly', () => {
        expect(getFilter()).toMatchSnapshot();
    });
});
