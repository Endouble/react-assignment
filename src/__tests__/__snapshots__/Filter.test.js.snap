// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Unit Test Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Filter
    onFilter={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Header
          as="h3"
        >
          Filters
        </Header>,
        <Form
          as="form"
          onSubmit={[Function]}
          role="search"
        >
          <FormGroup
            widths="equal"
          >
            <FormField>
              <label
                htmlFor="beerName"
              >
                Name
              </label>
              <Input
                id="beerName"
                placeholder="Name"
                type="text"
              />
            </FormField>
            <FormField>
              <label
                htmlFor="beerIbuGt"
              >
                IBU Greater Than
              </label>
              <Input
                id="beerIbuGt"
                onKeyPress={[Function]}
                placeholder="IBU Greater Than"
                type="number"
              />
            </FormField>
            <FormField>
              <label
                htmlFor="beerAbvGt"
              >
                ABV Greater Than
              </label>
              <Input
                id="beerAbvGt"
                onKeyPress={[Function]}
                placeholder="ABV Greater Than"
                type="number"
              />
            </FormField>
            <FormField
              className="form-action-container"
            >
              <Button
                as="button"
                primary={true}
                role="button"
              >
                Filter
              </Button>
            </FormField>
          </FormGroup>
        </Form>,
      ],
      "className": "filters-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "h3",
          "children": "Filters",
        },
        "ref": null,
        "rendered": "Filters",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "as": "form",
          "children": <FormGroup
            widths="equal"
          >
            <FormField>
              <label
                htmlFor="beerName"
              >
                Name
              </label>
              <Input
                id="beerName"
                placeholder="Name"
                type="text"
              />
            </FormField>
            <FormField>
              <label
                htmlFor="beerIbuGt"
              >
                IBU Greater Than
              </label>
              <Input
                id="beerIbuGt"
                onKeyPress={[Function]}
                placeholder="IBU Greater Than"
                type="number"
              />
            </FormField>
            <FormField>
              <label
                htmlFor="beerAbvGt"
              >
                ABV Greater Than
              </label>
              <Input
                id="beerAbvGt"
                onKeyPress={[Function]}
                placeholder="ABV Greater Than"
                type="number"
              />
            </FormField>
            <FormField
              className="form-action-container"
            >
              <Button
                as="button"
                primary={true}
                role="button"
              >
                Filter
              </Button>
            </FormField>
          </FormGroup>,
          "onSubmit": [Function],
          "role": "search",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <FormField>
                <label
                  htmlFor="beerName"
                >
                  Name
                </label>
                <Input
                  id="beerName"
                  placeholder="Name"
                  type="text"
                />
              </FormField>,
              <FormField>
                <label
                  htmlFor="beerIbuGt"
                >
                  IBU Greater Than
                </label>
                <Input
                  id="beerIbuGt"
                  onKeyPress={[Function]}
                  placeholder="IBU Greater Than"
                  type="number"
                />
              </FormField>,
              <FormField>
                <label
                  htmlFor="beerAbvGt"
                >
                  ABV Greater Than
                </label>
                <Input
                  id="beerAbvGt"
                  onKeyPress={[Function]}
                  placeholder="ABV Greater Than"
                  type="number"
                />
              </FormField>,
              <FormField
                className="form-action-container"
              >
                <Button
                  as="button"
                  primary={true}
                  role="button"
                >
                  Filter
                </Button>
              </FormField>,
            ],
            "widths": "equal",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <label
                    htmlFor="beerName"
                  >
                    Name
                  </label>,
                  <Input
                    id="beerName"
                    placeholder="Name"
                    type="text"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Name",
                    "htmlFor": "beerName",
                  },
                  "ref": null,
                  "rendered": "Name",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "id": "beerName",
                    "placeholder": "Name",
                    "type": "text",
                  },
                  "ref": Object {
                    "current": null,
                  },
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <label
                    htmlFor="beerIbuGt"
                  >
                    IBU Greater Than
                  </label>,
                  <Input
                    id="beerIbuGt"
                    onKeyPress={[Function]}
                    placeholder="IBU Greater Than"
                    type="number"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "IBU Greater Than",
                    "htmlFor": "beerIbuGt",
                  },
                  "ref": null,
                  "rendered": "IBU Greater Than",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "id": "beerIbuGt",
                    "onKeyPress": [Function],
                    "placeholder": "IBU Greater Than",
                    "type": "number",
                  },
                  "ref": Object {
                    "current": null,
                  },
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <label
                    htmlFor="beerAbvGt"
                  >
                    ABV Greater Than
                  </label>,
                  <Input
                    id="beerAbvGt"
                    onKeyPress={[Function]}
                    placeholder="ABV Greater Than"
                    type="number"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "ABV Greater Than",
                    "htmlFor": "beerAbvGt",
                  },
                  "ref": null,
                  "rendered": "ABV Greater Than",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "id": "beerAbvGt",
                    "onKeyPress": [Function],
                    "placeholder": "ABV Greater Than",
                    "type": "number",
                  },
                  "ref": Object {
                    "current": null,
                  },
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Button
                  as="button"
                  primary={true}
                  role="button"
                >
                  Filter
                </Button>,
                "className": "form-action-container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "as": "button",
                  "children": "Filter",
                  "primary": true,
                  "role": "button",
                },
                "ref": null,
                "rendered": "Filter",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Header
            as="h3"
          >
            Filters
          </Header>,
          <Form
            as="form"
            onSubmit={[Function]}
            role="search"
          >
            <FormGroup
              widths="equal"
            >
              <FormField>
                <label
                  htmlFor="beerName"
                >
                  Name
                </label>
                <Input
                  id="beerName"
                  placeholder="Name"
                  type="text"
                />
              </FormField>
              <FormField>
                <label
                  htmlFor="beerIbuGt"
                >
                  IBU Greater Than
                </label>
                <Input
                  id="beerIbuGt"
                  onKeyPress={[Function]}
                  placeholder="IBU Greater Than"
                  type="number"
                />
              </FormField>
              <FormField>
                <label
                  htmlFor="beerAbvGt"
                >
                  ABV Greater Than
                </label>
                <Input
                  id="beerAbvGt"
                  onKeyPress={[Function]}
                  placeholder="ABV Greater Than"
                  type="number"
                />
              </FormField>
              <FormField
                className="form-action-container"
              >
                <Button
                  as="button"
                  primary={true}
                  role="button"
                >
                  Filter
                </Button>
              </FormField>
            </FormGroup>
          </Form>,
        ],
        "className": "filters-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "h3",
            "children": "Filters",
          },
          "ref": null,
          "rendered": "Filters",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "form",
            "children": <FormGroup
              widths="equal"
            >
              <FormField>
                <label
                  htmlFor="beerName"
                >
                  Name
                </label>
                <Input
                  id="beerName"
                  placeholder="Name"
                  type="text"
                />
              </FormField>
              <FormField>
                <label
                  htmlFor="beerIbuGt"
                >
                  IBU Greater Than
                </label>
                <Input
                  id="beerIbuGt"
                  onKeyPress={[Function]}
                  placeholder="IBU Greater Than"
                  type="number"
                />
              </FormField>
              <FormField>
                <label
                  htmlFor="beerAbvGt"
                >
                  ABV Greater Than
                </label>
                <Input
                  id="beerAbvGt"
                  onKeyPress={[Function]}
                  placeholder="ABV Greater Than"
                  type="number"
                />
              </FormField>
              <FormField
                className="form-action-container"
              >
                <Button
                  as="button"
                  primary={true}
                  role="button"
                >
                  Filter
                </Button>
              </FormField>
            </FormGroup>,
            "onSubmit": [Function],
            "role": "search",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <FormField>
                  <label
                    htmlFor="beerName"
                  >
                    Name
                  </label>
                  <Input
                    id="beerName"
                    placeholder="Name"
                    type="text"
                  />
                </FormField>,
                <FormField>
                  <label
                    htmlFor="beerIbuGt"
                  >
                    IBU Greater Than
                  </label>
                  <Input
                    id="beerIbuGt"
                    onKeyPress={[Function]}
                    placeholder="IBU Greater Than"
                    type="number"
                  />
                </FormField>,
                <FormField>
                  <label
                    htmlFor="beerAbvGt"
                  >
                    ABV Greater Than
                  </label>
                  <Input
                    id="beerAbvGt"
                    onKeyPress={[Function]}
                    placeholder="ABV Greater Than"
                    type="number"
                  />
                </FormField>,
                <FormField
                  className="form-action-container"
                >
                  <Button
                    as="button"
                    primary={true}
                    role="button"
                  >
                    Filter
                  </Button>
                </FormField>,
              ],
              "widths": "equal",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <label
                      htmlFor="beerName"
                    >
                      Name
                    </label>,
                    <Input
                      id="beerName"
                      placeholder="Name"
                      type="text"
                    />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Name",
                      "htmlFor": "beerName",
                    },
                    "ref": null,
                    "rendered": "Name",
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "id": "beerName",
                      "placeholder": "Name",
                      "type": "text",
                    },
                    "ref": Object {
                      "current": null,
                    },
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <label
                      htmlFor="beerIbuGt"
                    >
                      IBU Greater Than
                    </label>,
                    <Input
                      id="beerIbuGt"
                      onKeyPress={[Function]}
                      placeholder="IBU Greater Than"
                      type="number"
                    />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "IBU Greater Than",
                      "htmlFor": "beerIbuGt",
                    },
                    "ref": null,
                    "rendered": "IBU Greater Than",
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "id": "beerIbuGt",
                      "onKeyPress": [Function],
                      "placeholder": "IBU Greater Than",
                      "type": "number",
                    },
                    "ref": Object {
                      "current": null,
                    },
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <label
                      htmlFor="beerAbvGt"
                    >
                      ABV Greater Than
                    </label>,
                    <Input
                      id="beerAbvGt"
                      onKeyPress={[Function]}
                      placeholder="ABV Greater Than"
                      type="number"
                    />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "ABV Greater Than",
                      "htmlFor": "beerAbvGt",
                    },
                    "ref": null,
                    "rendered": "ABV Greater Than",
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "id": "beerAbvGt",
                      "onKeyPress": [Function],
                      "placeholder": "ABV Greater Than",
                      "type": "number",
                    },
                    "ref": Object {
                      "current": null,
                    },
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <Button
                    as="button"
                    primary={true}
                    role="button"
                  >
                    Filter
                  </Button>,
                  "className": "form-action-container",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "as": "button",
                    "children": "Filter",
                    "primary": true,
                    "role": "button",
                  },
                  "ref": null,
                  "rendered": "Filter",
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
